omod/src/main/java/org/openmrs/module/webservices/rest/web/controller/SwaggerSpecificationController.java

```diff
@@ -27,45 +27,12 @@
 	public @ResponseBody
 	String getSwaggerSpecification(HttpServletRequest request) throws Exception {
 		
-		String swaggerSpecificationJSON = "";
-		StringBuilder baseUrl = new StringBuilder();
-		String scheme = request.getScheme();
-		int port = request.getServerPort();
-		
-		baseUrl.append(scheme); // http, https
-		baseUrl.append("://");
-		baseUrl.append(request.getServerName());
-		if ((scheme.equals("http") && port != 80) || (scheme.equals("https") && port != 443)) {
-			baseUrl.append(':');
-			baseUrl.append(request.getServerPort());
-		}
-		
-		baseUrl.append(request.getContextPath());
-		
-		String resourcesUrl = Context.getAdministrationService().getGlobalProperty(
-		    RestConstants.URI_PREFIX_GLOBAL_PROPERTY_NAME, baseUrl.toString());
-		
-		if (!resourcesUrl.endsWith("/")) {
-			resourcesUrl += "/";
-		}
-		
-		resourcesUrl += "ws/rest";
-		
-		String urlWithoutScheme = "";
-		
-		/* Swagger appends scheme to urls, so we should remove it */
-		if (scheme.equals("http"))
-			urlWithoutScheme = resourcesUrl.replace("http://", "");
-		
-		else if (scheme.equals("https"))
-			urlWithoutScheme = resourcesUrl.replace("https://", "");
-		
-		SwaggerSpecificationCreator creator = new SwaggerSpecificationCreator(urlWithoutScheme);
```

Comment:
All this logic does nothing but build the baseUrl (urlWithoutScheme) string and then pass it to the SwaggerSpecificationCreator(..). As noted before generated string was invalid. Thus results in invalid host, and basePath in the spec. No there's no regression.